pkg/util/install/utils.go

```diff
@@ -6,40 +6,51 @@
 package install
 
 import (
-	"fmt"
-	"strings"
+	"github.com/enmasseproject/enmasse/pkg/apis/enmasse/v1beta1"
+	"github.com/enmasseproject/enmasse/pkg/util/images"
 
-	iotv1alpha1 "github.com/enmasseproject/enmasse/pkg/apis/iot/v1alpha1"
 	appsv1 "k8s.io/api/apps/v1"
 	corev1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
 )
 
-// Apply some default deployment values
-func ApplyDeploymentDefaults(deployment *appsv1.Deployment, component string, app string, name string) {
-	if deployment.Labels == nil {
-		deployment.Labels = make(map[string]string)
+func createDefaultLabels(labels map[string]string, component string, name string) map[string]string {
+
+	if labels == nil {
+		labels = make(map[string]string)
 	}
 
-	deployment.Labels["component"] = component
-	deployment.Labels["app"] = app
-	deployment.Labels["name"] = name
+	labels["component"] = component
+	labels["app"] = "enmasse"
+	labels["name"] = name
 
-	deployment.Spec.Selector = &v1.LabelSelector{
-		MatchLabels: map[string]string{
-			"component": component,
-			"app":       app,
-			"name":      name,
-		},
-	}
+	return labels
+}
+
+// Apply standard set of labels
+func ApplyDefaultLabels(meta *v1.ObjectMeta, component string, name string) {
+	meta.Labels = createDefaultLabels(meta.Labels, component, name)
+}
+
+// Apply some default service values
+func ApplyServiceDefaults(service *corev1.Service, component string, name string) {
+
+	ApplyDefaultLabels(&service.ObjectMeta, component, name)
+	service.Spec.Selector = createDefaultLabels(nil, component, name)
+
+}
+
+// Apply some default deployment values
+func ApplyDeploymentDefaults(deployment *appsv1.Deployment, component string, name string) {
```

Comment:
It would be nice to keep the existing components vs app distinction, because 'component' could be more generic than the 'name'. I.e. app: enmasse, component: router, name: router-xyz

Reply:
It allows allow you to collect which resources (services, deployments, volumes etc.) belong to a particular component

Reply:
I don't really understand what you want me to change here? Do you want the "app" parameter back, always setting it to "enmasse"?

Reply:
Ah, sorry. Again I didn't see the full context and somehow I thought component was the app parameter based on the usage. Please ignore my comment!